### PR TITLE
Pipe error

### PR DESCRIPTION
**Pull request by LaurinUB**
Fixed all pipe errors apart from interrupting a big amount of closing sockets.

### CODE REVIEW

File: src/Socket.hpp [hunk 1]
```diff
@@ -42,6 +42,7 @@ class Socket {
   double timeout_;
   bool data_written_;
   std::string response_;
+  // std::string request_;
```
Comment (philippschwarzmueller) [inline src/Socket.hpp:45]: i'm not sure why this is commented out instead of deleted - can we kick it?

File: src/TcpServer.cpp [hunk 2]
```diff
@@ -141,23 +146,16 @@ void TcpServer::run() {
...
-    if (poll(this->pollfds_, this->numfds_ + 1, 100) == -1) {
+    if (poll(this->pollfds_, this->numfds_, 100) == -1) {
       perror("poll");
-      std::cout << this->numfds_ << std::endl;
       exitWithError("Poll failed");
     }
     checkSocketTimeout();
-    log("====== Waiting for a new connection ======\n\n\n");
-    checkUnfinished(this->sockets_);
+    // log("====== Waiting for a new connection ======\n\n\n");
```
Comment (philippschwarzmueller) [inline src/TcpServer.cpp:154]: this might also be deleted, right?

File: src/TcpServer.cpp [hunk 3]
```diff
@@ -193,13 +191,14 @@ void TcpServer::handleConnection(Socket& socket) {
   std::string stringyfied_buff(buffer);
   try {
     HTTPRequest req(stringyfied_buff, socket);
-    std::cout << req << std::endl;
+    // std::cout << req << std::endl;
```
Comment (philippschwarzmueller) [inline src/TcpServer.cpp:194]: this debug print needs the PRINT check

**Review by philippschwarzmueller — COMMENTED** [review]
have a look at the minor comments i made, besides them LGTM and also worked on our not-yet CI-integrated Postman test collection

File: src/Socket.hpp (hunk 1 above)
Comment (LaurinUB) [reply src/Socket.hpp:45]: Hole main loop is still to be updated. And request_ is a reminder to be added.

File: src/TcpServer.cpp (hunk 2 above)
Comment (LaurinUB) [reply src/TcpServer.cpp:154]: still a debug message that sprouted from my messy mind I'm sorry

File: src/TcpServer.cpp (hunk 3 above)
Comment (LaurinUB) [reply src/TcpServer.cpp:194]: All still WIP 🥇

**Review by philippschwarzmueller — APPROVED** [review]
Lets gooooo 🚀

**Pull request merged**